pkg/partials/partial.go

```diff
@@ -0,0 +1,34 @@
+package partials
+
+import (
+	"encoding/json"
+	"log"
+)
+
+// GetErrorReply returns an error reply for the given payload and error,
+// in the correct format required by the request type.
+func GetErrorReply(payloadBz []byte, err error) ([]byte, error) {
+	log.Printf("DEBUG: Partially Unmarshalling request: %s", string(payloadBz))
+	partialRequest, er := partiallyUnmarshalRequest(payloadBz)
+	if er != nil {
+		return nil, er
+	}
+	return partialRequest.GenerateErrorPayload(err)
+}
+
+// partiallyUnmarshalRequest unmarshals the payload into a partial request
+// that contains only the fields necessary to generate an error response and
+// handle accounting for the request's method.
+func partiallyUnmarshalRequest(payloadBz []byte) (PartialPayload, error) {
+	// First attempt to unmarshal the payload into a partial JSON-RPC request
+	var jsonReq partialJSONPayload
```

Comment:
Can you move json marshal / unmarshal into a `partiallyUnmarshalJsonRPCRequest` helper so it'll set a pattern to add other types?